[NO-TICKET] Update documentation to clarify what test mode does (#4182)
DataDog/dd-trace-rb: docs/GettingStarted.md

After setting up, your services will appear on the [APM services page](https://app.datadoghq.com/apm/services) within a few minutes. Learn more about [using the APM UI][visualization docs].

### Disabling during testing/specs and test optimization for Ruby

Tracing is enabled by default; this includes:

* When loading using `datadog/auto_instrument`
* When activating integrations using `tracing.instrument` in the `Datadog.configure` block
* When calling the `Datadog::Tracing.trace` method

If you'd like to disable tracing when running your tests/specs, you can set the `DD_TRACE_ENABLED` environment variable to `false` or via code:

```ruby
Datadog.configure do |c|
  c.tracing.enabled = false
end
```

> Furthermore, if you're trying to get more visibility into your tests or you're struggling with slow or flaky test suites, consider looking at Datadog's [Test optimization](https://docs.datadoghq.com/tests/) via the [`datadog-ci`](https://github.com/DataDog/datadog-ci-rb?tab=readme-ov-file#datadog-test-optimization-for-ruby) gem.

Reviewer: thanks for mentioning the test optimization tool!

Author: Seemed very relevant to mention is folks are searching for "spec"/"test" in our readme :)

#### Transporting in Test Mode

> The test mode can be used for testing distributed tracing behavior. Enabling this setting will mean every
> span will be sent (no sampling is applied) and the transport will send data synchronously.
> It is configured by setting `c.tracing.test_mode.enabled` to true.

Reviewer 2: ```suggestion
Use the test mode for testing distributed tracing behavior. Enabling this setting causes every
span to be sent (no sampling is applied) and causes the transport to send data synchronously.
To configure test mode, set `c.tracing.test_mode.enabled` to `true`.
```

> If you're looking to disable tracing while your test suite is running, consider using the `tracing.enabled` setting
> to disable sending of traces.

Reviewer 3: A minor thing, but it would be cool to give an example on how-to, even if it's obvious

```suggestion
If you're looking to disable tracing while your test suite is running, consider using the `tracing.enabled` setting
to disable sending of traces.

```ruby
Datadog.configure do |c|
  c.tracing.test_mode.enabled = false
end
```
```

Author: Wait, I'm somewhat confused with this suggestion -- did you mean to suggest an example of using `tracing.enable`?

I explicitly removed the example with `test_mode.enabled` and stated why (item 4) so I'm not sure I understand your suggestion :eyes:
